lib/ts/recipe/oauth2provider/api/revokeToken.ts

```diff
@@ -0,0 +1,50 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { send200Response, sendNon200ResponseWithMessage } from "../../../utils";
+import { APIInterface, APIOptions } from "..";
+import { UserContext } from "../../../types";
+
+export default async function revokeTokenPOST(
+    apiImplementation: APIInterface,
+    options: APIOptions,
+    userContext: UserContext
+): Promise<boolean> {
+    if (apiImplementation.revokeTokenPOST === undefined) {
+        return false;
+    }
+
+    const body = await options.req.getFormData();
+
+    if (body.token === undefined) {
+        sendNon200ResponseWithMessage(options.res, "token is required in the request body", 400);
+        return true;
+    }
+
+    const authorizationHeader =
+        options.req.getHeaderValue("authorization") || options.req.getHeaderValue("Authorization");
+
+    let response = await apiImplementation.revokeTokenPOST({
+        options,
+        authorizationHeader,
+        token: body.token,
+        clientId: body.client_id,
+        clientSecret: body.client_secret,
+        userContext,
+    });
+
+    send200Response(options.res, response);
```

Comment:
Don't we need to set the `WWW-Authorize` header in the response? What happens if the auth params are missing?

Reply:
Decided not to set it as couldn't find it anywhere in spec.